### PR TITLE
Devise token auth

### PR DESCRIPTION
**Pull request by fabianlundgren**
**PT Story:** https://www.pivotaltracker.com/story/show/144082759

**Changes proposed in this PR:**
- Adds devise_token_auth
- Changes Admin class to User
- Adds tenant, admin and janitor role.
- Changes root path to render sign up or log in when not logged in.

**What I have learned working on this feature:**
- Setting up custom validations.
- How to implement devise and devise_token_auth.
- How to work with devise routes.
- How to write request specs for registration.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :email }
+  end
+
+  describe 'Custom Validations' do
+    it 'has a default user role of tenant' do
+      expect(subject.role).to eq 'tenant'
+    end
+
+    it 'has a user role of clown' do
```
Comment (AmberWilkie) [inline spec/models/user_spec.rb]: `it 'cannot set a non-approved role'`

File: features/step_definitions/user_sign_up_steps.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Given(/^I click link "([^"]*)"$/) do |link|
+  click_link link
+end
+
+Given(/^I am on the registration page$/) do
+  visit new_user_registration_path
+end
+
+Then(/^I select "([^"]*)" from "([^"]*)"$/) do |option, role|
```
Comment (AmberWilkie) [inline features/step_definitions/user_sign_up_steps.rb]: make these more generic so your colleagues will understand what is going on here - something like `option, choices`

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -21,3 +21,18 @@
 Given(/^I should not see "([^"]*)"$/) do |content|
   expect(page).not_to have_content content
 end
+
+Given(/^I am on the index page$/) do
+  visit root_path
+end
+
+Given(/^the following user exist$/) do |table|
```
Comment (AmberWilkie) [inline features/step_definitions/basic_steps.rb]: `the following users exist`

**Review by AmberWilkie — COMMENTED** [review]
Ok just a couple little things. And it's not a blocker, because it's already done, but Devise and devise_token_auth are two different things that should have been done in two separate features.

**Comment by fottander** [comment]
Fixed!

**Review by AmberWilkie — APPROVED** [review]

**Comment by fottander** [comment]
We actually had a devise pr that got merged but a bit too early so we had to change some stuff for this feature to work with devise token auth.

**Pull request merged**